app/middleware/performPlaceSearch.js

```diff
@@ -0,0 +1,45 @@
+const log = require('../lib/logger');
+const locate = require('../lib/locate');
+const sortByLocalType = require('../lib/sortByLocalType');
+const routeHelper = require('./routeHelper');
+const renderer = require('./renderer');
+const createPlaceViewModel = require('./createPlaceViewModel');
+const placeSearches = require('../lib/promCounters').placeSearches;
+const placeDisambiguationViews = require('../lib/promCounters').placeDisambiguationViews;
+const zeroPlaceResultsViews = require('../lib/promCounters').zeroPlaceResultsViews;
+
+function logZeroResults(places, location) {
+  if (places.length === 0) {
+    log.warn({ location }, `No results were found for ${location}`);
+    zeroPlaceResultsViews.inc(1);
+  }
+}
+
+function getCoordinates(place) {
+  return {
+    latitude: place.latitude,
+    longitude: place.longitude
+  };
+}
+
+async function getPlaces(req, res, next) {
+  const location = req.query.location;
+  try {
+    let places = await locate.byPlace(location);
+    placeSearches.inc(1);
+    places = sortByLocalType(places.filter(place => place.country === 'England'));
+    logZeroResults(places, location);
+    if (places.length === 1) {
+      res.locals.coordinates = getCoordinates(places[0]);
+      next();
+    } else {
+      placeDisambiguationViews.inc(1);
+      res.locals.places = createPlaceViewModel(places);
+      renderer.places(req, res);
+    }
+  } catch (ex) {
+    routeHelper.renderFindHelpPage(req, res, location, 'Find places error');
```

Comment:
Should this render the find help page or an error page?
Looking at the equivalent for postcode lookup an error page would be returned.
The error gets logged too https://github.com/nhsuk/connecting-to-services/pull/369/files#diff-d2462ba736f50bc9398bf5779575bce9R23

Reply:
Changed to log an error and return the error page.